core/tee/tee_cryp_hkdf.c

```diff
@@ -137,7 +137,7 @@ static TEE_Result hkdf_expand(uint32_t hash_id, const uint8_t *prk,
 	}
 
 out:
-	free(ctx);
+	crypto_mac_free_ctx(ctx, hmac_algo);
```

Comment:
OK, this can go in a patch with subject: "Call `crypto_mac_free_ctx()` instead of `free()`"

Reply:
Hi @jforissier 
Thanks for your nice explanation.
As you said before, crypto_authenc_free_ctx() should be called only if crypto_authenc_alloc() succeeded. But I see in core/tee/tee_cryp_hkdf.c, it goto out when alloc fails and even before alloc.
https://github.com/OP-TEE/optee_os/blob/b924c494920f9197b4b6c9fc313b174306282f7b/core/tee/tee_cryp_hkdf.c#L76-L90

Reply:
This is sadly an error introduced when the crypto API was changed.

Reply:
Yeah `free()` should be `crypto_mac_free_ctx()` but please note that `free(NULL)` and similarly `crypto_mac_free_ctx(NULL, algo)` *are* valid.

Reply:
Hrm, @jforissier not entirely true, see:
https://github.com/OP-TEE/optee_os/blob/master/core/crypto/crypto.c#L103-L106

Reply:
Ok, I will update my patch after you send a PR

Reply:
https://github.com/OP-TEE/optee_os/pull/2134